src/lib/retry.sh

```diff
@@ -8,31 +8,27 @@ set -euo pipefail
 #
 # Successive backoffs double the timeout.
 function retry {
+  local n=1
   local max_attempts=${ATTEMPTS:-5}
   local timeout=${TIMEOUT:-1}
   local attempt=1
   local exitCode=0
 
-  while (( attempt <= max_attempts ))
-  do
-    if "$@"
-    then
-      return 0
-    fi
-    exitCode=$?
-
-    >&2 echo "Attempt #$attempt/$max_attempts failed."
-    attempt=$(( attempt + 1 ))
-    (( attempt > max_attempts )) && break
-
-    echo "Retrying in $timeout seconds ..."
-    sleep "$timeout"
-    timeout=$(( timeout * 2 ))
+  while true; do
+    "$@" && break || {
```

Comment:
As shellcheck notes: this is error-prone:
```
"$@" && break || {
         ^-- SC2015: Note that A && B || C is not if-then-else. C may run when A is true.
```

Won't happen in this instance as break exits the loop and won't return true, but still a pattern to avoid